shared/django_apps/codecov_auth/models.py

```diff
@@ -151,6 +159,24 @@ def __str__(self):
         str_representation_of_is_active = "Active" if self.is_active else "Inactive"
         return f"{str_representation_of_is_active} Account: {self.name}"
 
+    @property
+    def activated_user_count(self) -> int:
+        # Activated users include all _non_ students. So we'll need to get all the
+        # students and filter them out.
+        return self.users.filter(owners__student=False).count()
+
+    def can_activate_user(self, user: User | None) -> bool:
```

Comment:
just curious, when would user be None?

Reply:
ahhh okay gotcha. In the event user is None would we return false? I think now it will fall back to the bottom logic checking seats

Was thinking something like 

`
if not user or user in self.users.all():
    return False

if user.is_github_student:
    return True
`

Reply:
> In the event user is None would we return false?

Nope! The only reason to pass in the user is to check that they've already been added. If there was no user object associated, then it implicitly means that they haven't already been added to the account (and hence, we should then see if we can add them via the seat count check).

I guess having `User` be None is not super clear that it means that the owner is not already added to the account. I've considered maybe just removing the user-account link check but I felt like having it here was more convenient and more elegant in the client code (i.e specifically [here](https://github.com/codecov/worker/pull/556/files#diff-707fe54f561ef674e783cc52d7d93cc0e5760e9cb744b11d73776926e68d39b8R42)). Not sure if you have a strong opinion on one way or the other.

Reply:
HMmmMmmmm I don't have too much of a preference tbh

Maybe a small comment for when user could be None should be sufficient, especially since that seems like a legacy concept we shouldn't run into much / at all over time. 

Even that seems a little overkill though 😂

Reply:
Hmm... thinking about it a bit more, I think that we should make this optional. We should be able to call `can_activate_user()` with no params (as it shouldn't matter if the function has a user or not... we're concerned about seat count). I'll also add a comment.

Reply:
That makes sense to me! I think it's a lot more clear that way too